lib/pdk/module/release.rb

```diff
@@ -0,0 +1,260 @@
+require 'pdk'
+
+module PDK
+  module Module
+    class Release
+      def self.invoke(module_path, options = {})
+        new(module_path, options).run
+      end
+
+      attr_reader :options
+
+      attr_reader :module_path
+
+      def initialize(module_path, options = {})
+        @options = options
+
+        # TODO: Currently the release process can ONLY be run if the working directory IS the module root. However, in the future
+        # this WILL change, so we have the API arguments for it, but only accept `nil` for the first parameter
+        raise PDK::CLI::ExitWithError, _('Running the release process outside of the working directory is not supported') unless module_path.nil?
+
+        if module_path.nil?
+          module_path = PDK::Util.module_root
+          raise PDK::CLI::ExitWithError, _('The module release process requires a valid module path') % { module_path: module_path } if module_path.nil?
+        end
+        raise PDK::CLI::ExitWithError, _('%{module_path} is not a valid module') % { module_path: module_path } unless PDK::Util.in_module_root?(module_path)
+        @module_path = module_path
+      end
+
+      def run
+        # Pre-release checks
+        unless force?
+          raise PDK::CLI::ExitWithError, _('The module is not PDK compatible') if requires_pdk_compatibility? && !pdk_compatible?
+          raise PDK::CLI::ExitWithError, _('The module is not Forge compatible') if requires_forge_compatibility? && !forge_compatible?
+        end
+
+        # Note that these checks are duplicated in the run_publish method, however it's a much better
+        # experience to fail early, than going through the whole process, only to error at the end knowing full well
+        # it'll fail anyway.
+        validate_publish_options!
+
+        run_validations(options) unless skip_validation?
+
+        PDK.logger.info _('Releasing %{module_name} - from version %{module_version}') % {
+          module_name:    module_metadata.data['name'],
+          module_version: module_metadata.data['version'],
+        }
+
+        PDK::Util::ChangelogGenerator.generate_changelog unless skip_changelog?
+
+        # Calculate the new module version
+        new_version = specified_version
+        if new_version.nil? && !skip_changelog?
+          new_version = PDK::Util::ChangelogGenerator.compute_next_version(module_metadata.data['version'])
+        end
+        new_version = module_metadata.data['version'] if new_version.nil?
+
+        if new_version != module_metadata.data['version']
```

Comment:
Could this be tied to a `--force` type flag? It seems more likely that someone typos the version (1.0.1 instead of 1.1.0) and automatically changing it without specific acknowledgement of the difference could lead to undesired results.

Reply:
There is an implicit `force` in play here.  The user _has_ to specify `--version <...>` on the command line in order for that to happen.  Adding a requirement for `--force` as well seems unnecessary as this 2nd command line argument is not intuitive and gives no extra safety.